homeassistant/components/homematic/entity.py

```diff
@@ -130,24 +130,36 @@ def _hm_event_callback(self, device, caller, attribute, value):
 
     def _subscribe_homematic_events(self):
         """Subscribe all required events to handle job."""
-        for metadata in (
-            self._hmdevice.SENSORNODE,
-            self._hmdevice.BINARYNODE,
-            self._hmdevice.ATTRIBUTENODE,
-            self._hmdevice.WRITENODE,
-            self._hmdevice.EVENTNODE,
+        # Ordered list of nodes to determine priority when subscribing to
+        # events. Only one channel per parameter should be subscribed to
+        # to avoid jumping values.
+        # E.g., if a sensor node is given, that one should be used.
+        # Highest priority is on the bottom
+        node_priority = [
             self._hmdevice.ACTIONNODE,
-        ):
-            for node, channels in metadata.items():
-                # Data is needed for this instance
-                if node in self._data:
-                    # chan is current channel
-                    if len(channels) == 1:
-                        channel = channels[0]
-                    else:
-                        channel = self._channel
-                    # Remember the channel for this attribute to ignore invalid events later
-                    self._channel_map.add(f"{node}:{channel!s}")
+            self._hmdevice.EVENTNODE,
+            self._hmdevice.WRITENODE,
+            self._hmdevice.ATTRIBUTENODE,
+            self._hmdevice.BINARYNODE,
+            self._hmdevice.SENSORNODE,
+        ]
+
+        def _channel_list_to_single_channel(channel_list):
+            if len(channel_list) == 1:
+                return channel_list[0]
+            return self._channel
+
+        channel_dict = {
+            node: _channel_list_to_single_channel(channels)
+            for metadata in node_priority
+            for node, channels in metadata.items()
+            if node in self._data  # Data is needed for this instance
+        }
```

Comment:
That's really hard to read and understand. Let's keep the old structure in order. I think it would make also sense to change the handling after config flow migration. This was my first integration and that is a long time ago now

Reply:
@pvizeli Sure, makes sense to me. I will close this PR for the time being. Thanks for the review!